File: web-client/src/lib/types/dm-tool-types/definitions/abilityScoreDefinition.ts
import type { SkillDefinition, SkillDefinitionDto } from "./skillDefinition";

export interface AbilityScoreDefinition{
    id: string;
    name: string;
    abbreviation: string;
    description: string;
    skillDefinitions: SkillDefinition[];
}

export interface AbilityScoreDefinitionDto {
    id: string;
    name: string;
    abbreviation: string;
    description: string;
    skillDefinitions: SkillDefinitionDto[];
}

File: server/src/Models/DMToolkitModels/Definitions/AbilityScoreDefinition.cs
namespace DMToolkit.API.Models.DMToolkitModels.Definitions;

public class AbilityScoreDefinition : IDefinition
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Name { get; set; } = string.Empty;
    public string Abbreviation { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public ICollection<SkillDefinition> SkillDefinitions { get; set; } = new List<SkillDefinition>();
    public ICollection<BackgroundDefinition> BackgroundDefinitions { get; set; } = new List<BackgroundDefinition>();
}